### ISSUE TITLE
DL3018 Fails when packages are specified from an ARG

### ISSUE THREAD

**Issue by ianfixes** [hadolint/hadolint#393]
- [x] This is a bug report
- [x] I searched existing issues before opening this one (and #219 came close)

### Expected behavior
Don't trigger rule `DL3018` when the "package" starts with `$`, unless you can verify that the value of the expanded variable does indeed 

### Actual behavior
Hadolint apparently thinks `$BUILD_PACKAGES` is itself a package name

### Steps to reproduce the behavior
 
`test.Dockerfile`
```Dockerfile
FROM alpine:3
ARG BUILD_PACKAGES="bash=5.0.0-r0"
RUN apk add --update --no-cache $BUILD_PACKAGES
```

```terminal
$ docker build -f test.Dockerfile .
Sending build context to Docker daemon  722.4kB
Step 1/3 : FROM alpine:3
 ---> 961769676411
Step 2/3 : ARG BUILD_PACKAGES="bash=5.0.0-r0"
 ---> Running in 2aa40ead8d75
Removing intermediate container 2aa40ead8d75
 ---> 9183b2a76f42
Step 3/3 : RUN apk add --update --no-cache $BUILD_PACKAGES
 ---> Running in c263e750f229
fetch http://dl-cdn.alpinelinux.org/alpine/v3.10/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.10/community/x86_64/APKINDEX.tar.gz
(1/5) Installing ncurses-terminfo-base (6.1_p20190518-r0)
(2/5) Installing ncurses-terminfo (6.1_p20190518-r0)
(3/5) Installing ncurses-libs (6.1_p20190518-r0)
(4/5) Installing readline (8.0.0-r0)
(5/5) Installing bash (5.0.0-r0)
Executing bash-5.0.0-r0.post-install
Executing busybox-1.30.1-r2.trigger
OK: 15 MiB in 19 packages
Removing intermediate container c263e750f229
 ---> 06af8bdf97d1
Successfully built 06af8bdf97d1

$ docker run --rm -i hadolint/hadolint:latest /bin/hadolint --version
Haskell Dockerfile Linter v1.17.0-0-g56fd3e1

$ docker run --rm -i hadolint/hadolint:latest /bin/hadolint - < test.Dockerfile 
/dev/stdin:3 DL3018 Pin versions in apk add. Instead of `apk add <package>` use `apk add <package>=<version>`
```
